### ISSUE TITLE
Boltz Rescue Key: Recover all swaps using a single key file

### ISSUE THREAD

**Issue by michael1011** [BoltzExchange/boltz-web-app#91]

**Comment by armurbalda** [comment]
while discussing this a bit we realized that it might make sense to derive keys or even child seeds from an external key that the user already has and already took care of backing up, doing away with incognito mode, tor browser losing local storage and losing refund files

**Comment by kilrau** [comment]
If no alby connected -> Download refund file step pops up. Alby connected - nothing. Clean UX.

**Comment by kilrau** [comment]
This issue came up again today and we finally decided to go for the following:
- completely end the concept of single-use refund files
- do not rely on any external key managers like Alby
- instead force users to download a "Boltz Seed file" (first prompt download, then prompt "upload" to confirm it was downloaded) which contains a masterkey from which Boltz Web App will derive all future refund keys. If a Boltz Seed is already in the Browser's local db, there is no download step at all
- Boltz will supply the "metadata" like the swap tree via a new endpoint for users who refund using the seed file in a clean browser
- Tl;dr; we are adding a step for new users (download, safe seed file, upload to verify), users on a fresh browser (upload seed file) but taking a step away for all users who got a Boltz Seed in the browser already

Todo:
- new endpoint on backend taking a derived pubkey for a swap to refund, which will be the "primary key" to identify a swap in our db
- create wireframe for Web App flow and think a bit more about any potential gotchas, but testing should be easy as this is not a breaking change and can be tested in prod via a separate deployment
